openerp/addons/base/ir/ir_model.py

```diff
@@ -732,7 +732,7 @@ def group_names_with_access(self, cr, model_name, access_mode):
         return [('%s/%s' % x) if x[0] else x[1] for x in cr.fetchall()]
 
     @tools.ormcache()
-    def check(self, cr, uid, model, mode='read', raise_exception=True, context=None):
+    def _check(self, cr, uid, model, mode='read', raise_exception=True):
```

Comment:
When a cache miss happens, the context is still necessary to properly translate the error message.
The translation function does fallback to the user's language when no context is reachable, but wouldn't it be better to use `ormcache_context` to keep `context['lang']` in the cache key? (e.g. for cases where the user language might not be accurate, such as the `public` user)